src/typechecker/context-reduction.lisp

```diff
@@ -268,11 +268,20 @@ Returns (VALUES deferred-preds retained-preds defaultable-preds)"
         :do (setf subs (compose-substitution-lists subs (fundep-entail% env expr-preds pred known-tyvars)))
         :finally (return subs)))
 
+(defun expand-pred-into-superclasses (env pred)
```

Comment:
FYI: Beware that MAPCAN mutates the cons cells it gets, so always ensure you're receiving fresh conses/copies. Never MAPCAN constant data.

Reply:
Fixed in [f88d05d](https://github.com/coalton-lang/coalton/pull/1315/commits/f88d05dc50cc3279705315f20608e55718744533)